controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -1591,12 +1592,11 @@ private HttpResponse testConfig(ApplicationId id, JobType type) {
                                     .collect(Collectors.toSet());
         var testedZone = type.zone(controller.system());
 
-        // This should not happen, unless one asks for test-config for a production job, which one shouldn't have use for.
-        if (deployments.stream().anyMatch(deploymentId ->      deploymentId.zoneId().equals(testedZone)
-                                                          && ! deploymentId.applicationId().equals(id)))
-            throw new IllegalStateException("Conflict: " + testedZone + " contains both default and " + id.instance().value() + " instances");
+        // If a production job is specified, the production deployment of the _default instance_ is the relevant one,
```

Comment:
So tests are only run on the default instance because non-default instances are assumed to be tested implicitly by testing the default?

But someone set up only non-default instances?

Reply:
The assumption is that only the "default" instance is defined in deployment.xml for now, and config for a production job is therefore aimed at one of its deployments — not the user's `dev` instance, and not the user's non-existent production deployments :P Will need to expose instances in the test config API, but will do later.